utility_analysis_new/utility_analysis.py

```diff
@@ -102,27 +102,87 @@ def perform_utility_analysis(col,
     # accumulators : (aggregate_metrics)
 
     def pack_metrics(aggregate_metrics) -> List[metrics.AggregateMetrics]:
+        if options.multi_param_configuration is None:
```

Comment:
Yeah, that's a good interview task to simplify those computations :)

What do you think about the following:

1.`options.n_parameters`returns number of parameters for which Utility analysis is computed. 
2. we can assume that `aggregate_metrics` has `options.n_parameters` groups each corresponds to 1 configuration  (this will be the only assumption on format)
3. for those `i%options.n_parameters == 0`, create a new `metrics.AggregateMetrics()`
4. for all i use `_populate_packed_metrics` for the last created `metrics.AggregateMetrics`